HOWTO-HACK.txt

```diff
@@ -2,54 +2,74 @@ Instructions for folks who want to contribute to processing.py itself.
 
 Prerequisites:
 
-	git - http://git-scm.com/downloads
-	The latest JDK - http://www.oracle.com/technetwork/java/javase/downloads/index.html
-	Apache ant - http://ant.apache.org/bindownload.cgi
+    git - http://git-scm.com/downloads
+    JDK 8u144 - http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
+    Apache ant - http://ant.apache.org/bindownload.cgi
+
+JDK 8u144 is required until such time as Processing is built with JDK 9 or later.
 
 The Processing and Python Mode developers use Eclipse for their development,
 but it is not required. There are Eclipse project configs in our source trees.
 
-	Eclipse - https://www.eclipse.org/downloads/
+    Eclipse - https://www.eclipse.org/downloads/
 
-Steps:
+Steps
+-----
 
 Create a directory to organize your Processing-related source, say
 
-  mkdir processing-source
-  cd processing-source
+    $ mkdir processing-source
+    $ cd processing-source
+
+These build steps assume your file structure will look like this:
 
-If you think you're going to want to submit your own work to processing.py,
-you should start by creating a fork of processing.py on github. Let's say
+    processing-source/
+        |- processing.py/
+        |- processing/
+        |- processing-video/
+
+If you think you're going to want to submit your own work to Processing.py,
+you should start by creating a fork of Processing.py on GitHub. Let's say
 you've done so, and that your fork can be found at:
 
-  https://github.com/JRandomHacker/processing.py
+    https://github.com/JRandomHacker/processing.py
 
-I'll also assume you've uploaded your SSH public key to Github.  
+I'll also assume you've uploaded your SSH public key to GitHub.
 
-  $ git clone git@github.com:JRandomHacker/processing.py.git
+    $ git clone git@github.com:JRandomHacker/processing.py.git
 
-Otherwise, if you just want to build or examine Python Mode, you can
+Otherwise, if you just want to build or examine Python Mode, you can clone
+the original repo:
 
-  $ git clone https://github.com/jdf/processing.py.git
+    $ git clone https://github.com/jdf/processing.py.git
 
   
 You'll need source for processing and processing-video.
 
-  $ git clone --depth=1 https://github.com/processing/processing.git
-  $ git clone --depth=1 https://github.com/processing/processing-video.git
+    $ git clone --depth=1 https://github.com/processing/processing.git
+    $ git clone --depth=1 https://github.com/processing/processing-video.git
+
+Build processing core from source (according to the Processing build instructions):
```

Comment:
There's no reason to do this; the processing.py build should do it for you. Did it not?

Reply:
It did! I tried again on a clean slate, and it all built in one go. Will remove.